miio/integrations/vacuum/roborock/vacuum.py

```diff
@@ -361,6 +365,36 @@ def map(self):
         # returns ['retry'] without internet
         return self.send("get_map_v1")
 
+    @command()
+    def get_multi_maps(self, skip_cache=False) -> MultiMapList:
+        """Return list of multi maps."""
+        if self._multi_maps is not None and not skip_cache:
+            return self._multi_maps
+
+        self._multi_maps = MultiMapList(self.send("get_multi_maps_list")[0])
+        return self._multi_maps
+
+    @command()
+    def multi_map_enum(self, skip_cache=False) -> Optional[enum.Enum]:
+        """Enum of the available map names."""
```

Comment:
Yes that is nessesary, the get_multi_maps container is nicely fromatted and contains all information.
The multi_map_enum does not contain all information, but is nessesary for the setting decorator:
```
    @setting(
        "Multi map",
        choices_attribute="multi_map_enum",
        setter_name="load_multi_map_by_enum",
        icon="mdi:floor-plan",
    )
```
The decorator can only handle a function that directly returns the enum, it can not handle containers...